packages/react-reconciler/src/ReactProfileTimer.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from './ReactFiber';
+import type {StackCursor} from './ReactFiberStack';
+
+import warning from 'fbjs/lib/warning';
+import ReactFiberStack from './ReactFiberStack';
+
+const hasNativePerformanceNow =
+  typeof performance === 'object' && typeof performance.now === 'function';
+
+let now;
+if (hasNativePerformanceNow) {
+  now = function() {
+    return performance.now();
+  };
+} else {
+  now = function() {
+    return Date.now();
+  };
+}
+
+/**
+ * The "actual" render time is total time required to render the descendants of a Profiler component.
+ * This time is stored as a stack, since Profilers can be nested.
+ * This time is started during the "begin" phase and stopped during the "complete" phase.
+ * It is paused (and accumulated) in the event of an interruption or an aborted render.
+ */
+
+const {checkThatStackIsEmpty, createCursor, push, pop} = ReactFiberStack();
+
+let stackCursor: StackCursor<Fiber> = createCursor(null);
+let timerPausedAt: number = 0;
+let totalElapsedPauseTime: number = 0;
```

Comment:
This only increases but never gets reset. That seems odd. At some point we need to reset it right?

Reply:
Do we? Why? This time is accounted for in both the start and stop time methods.

I supposed I could reset it during the commit phase.